osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverter.java

```diff
@@ -36,30 +46,95 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GridMeasurementPublishedEventConverter.class);
 
-    private static final int SIMPLE_END_INDEX = 10;
+    private ObjectMapper objectMapper = new ObjectMapper();
 
     @Override
     public GridMeasurementPublishedEvent convert(final String source,
             final Type<? extends GridMeasurementPublishedEvent> destinationType, final MappingContext mappingContext) {
 
-        StringArrayToAnalogList stringArrayToAnalogList = null;
-        final String[] values = source.split(";");
-        if (values.length == SIMPLE_END_INDEX) {
-            stringArrayToAnalogList = new SimpleStringToAnalogList();
-        } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
-            stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
-        } else {
-            LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+        LsMeasurementMessageToAnalogList stringArrayToAnalogList = null;
+        LOGGER.debug("Source string: {}", source);
+
+        try {
+            final Payload[] payloads = this.objectMapper.readValue(source, Payload[].class);
+            if (payloads[0].equals(null)) {
```

Comment:
I meant the case where payloads.length == 0, that would result in the null pointer. You can't call equals on an array element with a null value. Since in that case, payload[0] references to null.